File: yu-ai-router-sdk/src/test/java/com/yupi/ai/sdk/example/StreamExample.java
/**
 * 流式调用示例
 *
 * @author <a href="https://codefather.cn">编程导航学习圈</a>
 */
package com.yupi.ai.sdk.example;

import com.yupi.ai.sdk.YuAIClient;
import com.yupi.ai.sdk.callback.StreamCallback;
import com.yupi.ai.sdk.model.ChatChunk;

/**
 * 流式调用示例
 */
public class StreamExample {

    public static void main(String[] args) throws InterruptedException {
        YuAIClient client = YuAIClient.builder()
                .apiKey("sk-d8483dd850cf44729f41cb59f7bc1e1d")
                .baseUrl("http://localhost:8123/api")
                .build();

        try {
            System.out.println("开始流式调用...\n");

            client.chatStream("写一首关于春天的诗", new StreamCallback() {
                @Override
                public void onMessage(ChatChunk chunk) {
                    // 处理深度思考内容
                        if (chunk.getReasoningContent() != null) {
                        System.out.println("[思考] " + chunk.getReasoningContent());
                    }
                    // 处理普通文本内容
                    if (chunk.getContent() != null) {
                        System.out.print(chunk.getContent());
                    }
                }

                @Override
                public void onComplete() {
                    System.out.println("\n\n✅ 完成");
                }

                @Override
                public void onError(Throwable error) {
                    System.err.println("\n❌ 错误: " + error.getMessage());
                    error.printStackTrace();
                }
            });

            // 等待流式响应完成
            Thread.sleep(10000);

        } finally {
            client.close();
        }
    }
}

File: yu-ai-router-python-sdk/examples/stream_example.py
"""Stream SDK usage example."""

from __future__ import annotations

import os

from yu_ai_router_sdk import ChatChunk, StreamCallback, YuAIClient


class PrintStreamCallback(StreamCallback):
    def on_message(self, chunk: ChatChunk) -> None:
        if chunk.reasoning_content:
            print(f"[思考] {chunk.reasoning_content}")
        if chunk.content:
            print(chunk.content, end="", flush=True)

    def on_complete(self) -> None:
        print("\n\n✅ 完成")

    def on_error(self, error: Exception) -> None:
        print(f"\n❌ 错误: {error}")


def main() -> None:
    api_key = os.getenv("YU_AI_API_KEY", "")
    base_url = os.getenv("YU_AI_BASE_URL", "http://localhost:8123/api")
    client = YuAIClient.builder().api_key(api_key).base_url(base_url).build()
    try:
        print("开始流式调用...\n")
        client.chat_stream("写一首关于春天的诗", PrintStreamCallback())
    finally:
        client.close()


if __name__ == "__main__":
    main()
